### PR TITLE
Split the unified PR template into few

### PR DESCRIPTION
**Pull request by webknjaz**
Ref: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository

##### SUMMARY

$sbj. This is not forms like for issues but the next best thing.

##### ISSUE TYPE

- Maintenance Pull Request

##### ADDITIONAL INFORMATION

Demo of how this works in the test repo: https://github.com/ansible/issue-forms-demo-for-ansible/compare/main...pr-demo

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/01-bug-fix.md
Comment (mkrizek) [inline .github/PULL_REQUEST_TEMPLATE/01-bug-fix.md]: Bot already does this. Unless this is done for all the templates, issues included, the bot would have to do the work anyway.

File: .github/PULL_REQUEST_TEMPLATE/01-bug-fix.md
Comment (mkrizek) [inline .github/PULL_REQUEST_TEMPLATE/01-bug-fix.md]: Same as above applies. Additionally this will break some of the bot's functionality as it relies on `needs_triage` not being present to identify the issue/PR as new/unseen. If we decide we want the `needs_triage` label be added this way, changes to the bot codebase need to be made before merging this.

**Review by mkrizek — DISMISSED** [review]
Looking at the changes I assume this does **_not_** change the format of the template **_after_** it is sent, is that correct?

I am not really fan of the labels being added automatically by this as the bot needs to account for that.

File: .github/PULL_REQUEST_TEMPLATE/01-bug-fix.md
Comment (webknjaz) [reply .github/PULL_REQUEST_TEMPLATE/01-bug-fix.md]: Totally. But this makes it a bit more responsive. I think it's a nice touch that could eventually lead to dropping this specific bit of classification from the bot. It's also more resilient since the submitters can't actually change these labels, meaning that the bot won't need to rely on heuristics of what text shows up in the issue body to know that the creator actually clicked that button for the correct category.

File: .github/PULL_REQUEST_TEMPLATE/01-bug-fix.md
Comment (webknjaz) [reply .github/PULL_REQUEST_TEMPLATE/01-bug-fix.md]: Ah, I didn't know. I thought it'd make getting the PRs into the triage queue faster. Can the bot be changed to have its logic recognize new submissions based on the timestamp?

**Comment by webknjaz** [comment]
> Looking at the changes I assume this does **_not_** change the format of the template **_after_** it is sent, is that correct?

@mkrizek the template shows up the same way, as a free-form text so the submitters are still able to change what's in the PR they create or delete it entirely. But they can't change the labels (unless they have enough privileges in the repo, of course).

File: .github/PULL_REQUEST_TEMPLATE/01-bug-fix.md
Comment (webknjaz) [reply .github/PULL_REQUEST_TEMPLATE/01-bug-fix.md]: FWIW I dropped `needs_triage` to prevent breaking bot's assumptions but kept the type labels for now.

**Comment by mkrizek** [comment]
> @mkrizek would it be fine if we move the addition of `needs_triage` to the follow-ups but keep the type-labels? I don't think it'll break the bot this way but would provide a nicer initial state.

It shouldn't break anything but:
> It's also more resilient since the submitters can't actually change these labels

will not apply because the bot will still parse the template and change the label based on it, so if the submitter goes back and changes the issue type, the bot will change the labels accordingly.

Ultimately I think that if we want labels being added by anyone or anything outside of the core team (commiters) or @ansibot then the change to the bot code is likely needed.

**Comment by webknjaz** [comment]
> will not apply because the bot will still parse the template and change the label based on it, so if the submitter goes back and changes the issue type, the bot will change the labels accordingly.

I meant though the GH UI, outside of the bot logic, it won't be changeable. So the initial value is set and then, only the bot will be updating it which is fine.

File: .github/PULL_REQUEST_TEMPLATE/05-fallback.md
Comment (s-hertel) [inline .github/PULL_REQUEST_TEMPLATE/05-fallback.md]: Can we omit title from the templates? It seems like people might not bother filling them out, leading to duplicates and very-similarly named PRs, making it more difficult to search for specific PRs. Generating one from the branch name or commit seems more helpful to me.

Comment (webknjaz) [reply .github/PULL_REQUEST_TEMPLATE/05-fallback.md]: Apparently, this metadata is not actually used by GH and the repo where I saw this dropped it in a follow-up: https://github.com/python/peps/pull/2998. So I'll delete these sections entirely too.

**Comment by webknjaz** [comment]
@mkrizek @s-hertel your concerns are now fully resolved.

File: .github/PULL_REQUEST_TEMPLATE/✨ New feature.md
Comment (flowerysong) [inline .github/PULL_REQUEST_TEMPLATE/✨ New feature.md]: This comment no longer makes sense.

Comment (webknjaz) [reply .github/PULL_REQUEST_TEMPLATE/✨ New feature.md]: Thanks for pointing this out!

Comment (webknjaz) [reply .github/PULL_REQUEST_TEMPLATE/✨ New feature.md]: Addressed. Marking as resolved.

**Comment by webknjaz** [comment]
It seems that the UI part is hidden by some feature toggle so I opened a ticket with GitHub to see what's up with that. Meanwhile, the old template behavior will remain preserved whenever. Once merged, the new templates will only be accessible by URL for the time being: https://github.com/ansible/issue-forms-demo-for-ansible/compare/pr-demo?template=Documentation%20change.md. Hopefully, GH will be able to fix it and it'll start to magically work.

FWIW I think this is mergeable even without their fix.

**Comment by webknjaz** [comment]
I'm going to merge this once the sanity tests pass post rebase.

**Pull request merged**
